src/hooks/useAllowance.ts

```diff
@@ -0,0 +1,95 @@
+import { MaxUint256 } from '@ethersproject/constants'
+import { TransactionResponse } from '@ethersproject/providers'
+import { Currency, CurrencyAmount, Token } from '@uniswap/sdk-core'
+import { useWeb3React } from '@web3-react/core'
+import { useTokenContract } from 'hooks/useContract'
+import { useTokenAllowance } from 'hooks/useTokenAllowance'
+import { useCallback, useMemo } from 'react'
+import { calculateGasMargin } from 'utils/calculateGasMargin'
+
+export enum AllowanceState {
+  UNKNOWN,
+  NOT_ALLOWED,
```

Comment:
allowances aren't binary, so I think this code might be a bit easier to follow if you use SUFFICIENT/INSUFFICIENT here instead of (NOT_)ALLOWED